### ISSUE TITLE
Mess in views

### ISSUE THREAD

**Issue by vryazanov** [timelesslounge/timelessis#173]
Hey, I looked at a few our packages with `views.py` and realized that we implemented them not the best way.
Basically they have:
1. `list`, `create`, `edit` and `delete` methods
2. with endpoints with the same naming (`/`, `/create`, `/edit` and etc)

My concern is that `function views` are not good for us since we are doing backend on templates. A bit later we will definitely have some "common" context to push to templates, and it's a way more easier to use `class based views`, we will be able to use mixins with common context (like user's info) and so on. Also currently, as I can see, we are copy-pasting a lot of code related to views . Ideally we should have some base class for CRUD operations, like
```python
class BaseTemplateView:
    def get_context(self):
         raise NotImplementedError
    def create(self):
         raise NotImplementedError
    def edit(self):
         raise NotImplementedError
    def delete(self):
         raise NotImplementedError
```
and some function which will register such views in router and map method names to endpoints automatically. We can go even further and implement `create` and `edit` methods in this class because the only difference between views will be the form class they are processing. Final solution may look like (like it's made in `django` framework):
```python
class CompanyView(BaseTemplateView):
    templates_path = 'templates/companies'
    form = CompanyForm
```
Also I recently received a puzzle #170 which forces me to use `API` endpoints. So looks like not all understand that we create backend with templates, not for single page application. (we already have `/api/companies/` and `/api/comments/` endpoints)

So I suggest:
1) Remove `API` endpoints in `Comments` and `Companies`
2) Implement simple base class for CRUD operations or find third-party solution

**Comment by 0crat** [comment]
@emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) please, pay attention to this issue

**Comment by emilianodellacasa** [comment]
@vryazanov Actually I wouln't remove API because we will need in future interactions. Please tell me if you would use a third party library or develop from scratch

**Comment by vryazanov** [comment]
@emilianodellacasa  we need to support CRUD operations for both template views and `API` views, right? Not my business, but I'm sure that implement SPA and keep frontend related code in separate repo will be more maintainable than keeping them in this repository and have a deal with copy-pasting in views.

As for your question, I think it's better to develop from scratch since flask provides powerful `class based views`, need just a bit customize them.

**Comment by 0crat** [comment]
Job [#173](https://github.com/timelesslounge/timelessis/issues/173) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
The job [#173](https://github.com/timelesslounge/timelessis/issues/173) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/b26a5785-40e8-4cdd-b10c-0b1b1d6de7d9); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/timelesslounge/timelessis/issues/173#issuecomment-462150838))

@paulodamaso The user @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) resigned from [#173](https://github.com/timelesslounge/timelessis/issues/173), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
The job [#173](https://github.com/timelesslounge/timelessis/issues/173) assigned to @bkuzmic[/z](https://www.0crat.com/u/bkuzmic), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/ef88f87a-bed2-4a9f-840d-f4e9f4c48c49); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@bkuzmic[/z](https://www.0crat.com/u/bkuzmic) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by bkuzmic** [comment]
@0crat wait for #287 to merge.

**Comment by 0crat** [comment]
> @0crat wait for #287 to merge.  ([here](https://github.com/timelesslounge/timelessis/issues/173#issuecomment-465118176))

@bkuzmic The impediment for [#173](https://github.com/timelesslounge/timelessis/issues/173) was registered successfully by @bkuzmic[/z](https://www.0crat.com/u/bkuzmic)

**Comment by 0pdd** [comment]
@vryazanov 8 puzzles [#305](https://github.com/timelesslounge/timelessis/issues/305), [#306](https://github.com/timelesslounge/timelessis/issues/306), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312) are still not solved.

**Comment by bkuzmic** [comment]
@vryazanov Please close the issue now that PR is merged.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @bkuzmic[/z](https://www.0crat.com/u/bkuzmic), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#173](https://github.com/timelesslounge/timelessis/issues/173) is now out of scope

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @bkuzmic[/z](https://www.0crat.com/u/bkuzmic)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0pdd** [comment]
@vryazanov 7 puzzles [#306](https://github.com/timelesslounge/timelessis/issues/306), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305).

**Comment by 0pdd** [comment]
@vryazanov 6 puzzles [#306](https://github.com/timelesslounge/timelessis/issues/306), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#307](https://github.com/timelesslounge/timelessis/issues/307).

**Comment by 0pdd** [comment]
@vryazanov 5 puzzles [#306](https://github.com/timelesslounge/timelessis/issues/306), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#310](https://github.com/timelesslounge/timelessis/issues/310).

**Comment by 0pdd** [comment]
@vryazanov 7 puzzles [#306](https://github.com/timelesslounge/timelessis/issues/306), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312), [#429](https://github.com/timelesslounge/timelessis/issues/429), [#430](https://github.com/timelesslounge/timelessis/issues/430) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#309](https://github.com/timelesslounge/timelessis/issues/309).

**Comment by 0pdd** [comment]
@vryazanov 7 puzzles [#306](https://github.com/timelesslounge/timelessis/issues/306), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#429](https://github.com/timelesslounge/timelessis/issues/429), [#430](https://github.com/timelesslounge/timelessis/issues/430), [#437](https://github.com/timelesslounge/timelessis/issues/437) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#312](https://github.com/timelesslounge/timelessis/issues/312).

**Comment by 0pdd** [comment]
@vryazanov 8 puzzles [#306](https://github.com/timelesslounge/timelessis/issues/306), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#429](https://github.com/timelesslounge/timelessis/issues/429), [#430](https://github.com/timelesslounge/timelessis/issues/430), [#437](https://github.com/timelesslounge/timelessis/issues/437), [#442](https://github.com/timelesslounge/timelessis/issues/442), [#443](https://github.com/timelesslounge/timelessis/issues/443) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#312](https://github.com/timelesslounge/timelessis/issues/312).

**Comment by 0pdd** [comment]
@vryazanov 7 puzzles [#308](https://github.com/timelesslounge/timelessis/issues/308), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#429](https://github.com/timelesslounge/timelessis/issues/429), [#430](https://github.com/timelesslounge/timelessis/issues/430), [#437](https://github.com/timelesslounge/timelessis/issues/437), [#442](https://github.com/timelesslounge/timelessis/issues/442), [#443](https://github.com/timelesslounge/timelessis/issues/443) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#306](https://github.com/timelesslounge/timelessis/issues/306), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#312](https://github.com/timelesslounge/timelessis/issues/312).

**Comment by 0pdd** [comment]
@vryazanov 7 puzzles [#308](https://github.com/timelesslounge/timelessis/issues/308), [#429](https://github.com/timelesslounge/timelessis/issues/429), [#430](https://github.com/timelesslounge/timelessis/issues/430), [#437](https://github.com/timelesslounge/timelessis/issues/437), [#442](https://github.com/timelesslounge/timelessis/issues/442), [#443](https://github.com/timelesslounge/timelessis/issues/443), [#475](https://github.com/timelesslounge/timelessis/issues/475) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#306](https://github.com/timelesslounge/timelessis/issues/306), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312).

**Comment by 0pdd** [comment]
@vryazanov 6 puzzles [#429](https://github.com/timelesslounge/timelessis/issues/429), [#430](https://github.com/timelesslounge/timelessis/issues/430), [#437](https://github.com/timelesslounge/timelessis/issues/437), [#442](https://github.com/timelesslounge/timelessis/issues/442), [#443](https://github.com/timelesslounge/timelessis/issues/443), [#475](https://github.com/timelesslounge/timelessis/issues/475) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#306](https://github.com/timelesslounge/timelessis/issues/306), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312).

**Comment by 0pdd** [comment]
@vryazanov 5 puzzles [#430](https://github.com/timelesslounge/timelessis/issues/430), [#437](https://github.com/timelesslounge/timelessis/issues/437), [#442](https://github.com/timelesslounge/timelessis/issues/442), [#443](https://github.com/timelesslounge/timelessis/issues/443), [#475](https://github.com/timelesslounge/timelessis/issues/475) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#306](https://github.com/timelesslounge/timelessis/issues/306), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312), [#429](https://github.com/timelesslounge/timelessis/issues/429).

**Comment by 0pdd** [comment]
@vryazanov 5 puzzles [#437](https://github.com/timelesslounge/timelessis/issues/437), [#442](https://github.com/timelesslounge/timelessis/issues/442), [#443](https://github.com/timelesslounge/timelessis/issues/443), [#475](https://github.com/timelesslounge/timelessis/issues/475), [#528](https://github.com/timelesslounge/timelessis/issues/528) are still not solved; solved: [#305](https://github.com/timelesslounge/timelessis/issues/305), [#306](https://github.com/timelesslounge/timelessis/issues/306), [#307](https://github.com/timelesslounge/timelessis/issues/307), [#308](https://github.com/timelesslounge/timelessis/issues/308), [#309](https://github.com/timelesslounge/timelessis/issues/309), [#310](https://github.com/timelesslounge/timelessis/issues/310), [#311](https://github.com/timelesslounge/timelessis/issues/311), [#312](https://github.com/timelesslounge/timelessis/issues/312), [#429](https://github.com/timelesslounge/timelessis/issues/429), [#430](https://github.com/timelesslounge/timelessis/issues/430).
